sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2685,9 +2685,12 @@ case class StringDecode(bin: Expression, charset: Expression)
   since = "1.5.0",
   group = "string_funcs")
 // scalastyle:on line.size.limit
-case class Encode(value: Expression, charset: Expression)
+case class Encode(value: Expression, charset: Expression, legacyCharsets: Boolean)
```

Comment:
Do we really need the new parameter? It seems we can use `SQLConf.get.legacyJavaCharsets` directly.

Reply:
This avoids some issues, see the motivation in the PR's description: https://github.com/apache/spark/pull/27658

Reply:
Got it.